plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py

```diff
@@ -82,47 +28,65 @@ class AutoQueueConf:
     # name of the job
     name: str = "${hydra.job.name}"
 
-    # following parameters are SLURM specific
 
+@dataclass
+class SlurmParams(BaseParams):
+    """Slurm configuration overrides and specific parameters
+    """
+
+    # Params are used to configure sbatch, for more info check:
+    # https://github.com/facebookincubator/submitit/blob/master/submitit/slurm/slurm.py
+
+    # Following parameters are slurm specific
+    # More information: https://slurm.schedmd.com/sbatch.html
+    #
+    # slurm partition to use on the cluster
+    partition: Optional[str] = None
+    comment: Optional[str] = None
+    constraint: Optional[str] = None
+    exclude: Optional[str] = None
+
+    # Following parameters are submitit specifics
+    #
+    # USR1 signal delay before timeout
+    signal_delay_s: int = 120
     # Maximum number of retries on job timeout.
     # Change this only after you confirmed your code can handle re-submission
     # by properly resuming from the latest stored checkpoint.
     # check the following for more info on slurm_max_num_timeout
     # https://github.com/facebookincubator/submitit/blob/master/docs/checkpointing.md
-    slurm_max_num_timeout: int = 0
-    # USR1 signal delay before timeout for the slurm queue
-    slurm_signal_delay_s: int = 30
-    # slurm partition to use on the cluster
-    slurm_partition: Optional[str] = None
+    max_num_timeout: int = 0
 
 
 @dataclass
-class QueueParams:
-    slurm: SlurmQueueConf = SlurmQueueConf()
-    local: LocalQueueConf = LocalQueueConf()
-    auto: AutoQueueConf = AutoQueueConf()
+class LocalParams(BaseParams):
+    pass
 
 
 @dataclass
-class SubmititConf:
-    queue: QueueType = QueueType.slurm
+class SlurmConf(ObjectConf):
+    cls: str = "hydra_plugins.hydra_submitit_launcher.submitit_launcher.SlurmSubmititLauncher"
+    params: SlurmParams = SlurmParams()
 
-    folder: str = "${hydra.sweep.dir}/.${hydra.launcher.params.queue}"
 
-    queue_parameters: QueueParams = QueueParams()
+@dataclass
+class LocalConf(ObjectConf):
+    cls: str = "hydra_plugins.hydra_submitit_launcher.submitit_launcher.LocalSubmititLauncher"
+    params: LocalParams = LocalParams()
 
 
-@dataclass
-class SubmititLauncherConf(ObjectConf):
-    cls: str = "hydra_plugins.hydra_submitit_launcher.submitit_launcher.SubmititLauncher"
-    params: SubmititConf = SubmititConf()
-    # memory to reserve for the job on each node, in GB
-    mem_limit: int = 2
+# finally, register two different choices:
+ConfigStore.instance().store(
+    group="hydra/launcher",
+    name="submitit_local",
+    node=LocalConf,
+    provider="submitit_launcher",
+)
```

Comment:
I think we can lose LocalConf and SlurmConf class definitions without losing functionality or clarity.
```
ConfigStore.instance().store(..., node=ObjectConf(cls: str ="XYZClass", params = XYZParams()))
```

Reply:
I may have done something wrong but I get:
omegaconf.errors.UnsupportedValueType: Value 'LocalParams' is not a supported primitive type

Reply:
Is LocalParama a @dataclass?

Reply:
its a BaseParams, which is a dataclass, does it need a decorator as well? I'm not really familiar with dataclasses

Reply:
Try to create a minimal example showing the problem with OmegaConf directly.
this seems to work and is very close to what you are doing.

```python
In [1]: from dataclasses import dataclass
In [9]: from typing import Any

In [2]: from omegaconf import OmegaConf

In [3]: @dataclass
   ...: class Foo:
   ...:     num : int = 10
   ...: 

In [4]: @dataclass
   ...: class LocalFoo(Foo):
   ...:     name: str = "name"
   ...: 

In [5]: OmegaConf.structured(LocalFoo)
Out[5]: {'num': 10, 'name': 'name'}

In [6]: OmegaConf.structured(LocalFoo())
Out[6]: {'num': 10, 'name': 'name'}

In [7]: OmegaConf.structured({"add-hoc":LocalFoo()})
Out[7]: {'add-hoc': {'num': 10, 'name': 'name'}}



In [10]: @dataclass
    ...: class ObjectConf:
    ...:     cls: str = "abc"
    ...:     params: Any = None
    ...: 

In [11]: OmegaConf.structured(ObjectConf(cls="xyz", params=LocalFoo()))
Out[11]: {'cls': 'xyz', 'params': {'num': 10, 'name': 'name'}}
```

Reply:
Example from someone else (he didn't complain so I think it's working for him):

https://github.com/anthonytec2/pytorch-lightning/pull/1/files/6b8d535b6eef983184b95beb3aec7838f6df41ab..09c393501a2c7ba451f2fc15e8d91b608fe2e770